mozilla_version/test/test_gecko.py

```diff
@@ -523,6 +523,44 @@ def test_gecko_version_ensures_a_new_added_release_type_is_caught(monkeypatch):
 def test_gecko_version_supports_old_schemes(version_string):
     assert str(GeckoVersion.parse(version_string)) == version_string
 
+
+@pytest.mark.parametrize('version_string, expected', (
+    ('31.0', '31.0esr'),
+    ('31.0build1', '31.0esrbuild1'),
+    ('31.0build2', '31.0esrbuild1'),
+    ('31.0.1', '31.0.1esr'),
+
+    ('32.0a1', '32.0a2'),
+    ('32.0a2', '32.0b1'),
+    ('32.0b1', '32.0'),
+    ('32.0b2', '32.0'),
+    ('32.0b10', '32.0'),
+    ('32.0b10', '32.0'),
+    ('32.0b10build3', '32.0build1'),
+
+    ('54.0a1', '54.0a2'),
+    ('54.0a2', '54.0b1'),
+    ('55.0a1', '55.0b1'),
+
+    ('2.0b2', '2.0rc1'),
+    ('2.0rc1', '2.0'),
+    ('2.0rc2', '2.0'),
```

Comment:
Should this test some past 54, so they're not covered by the check against `self._LAST_AURORA_DEVEDITION_AS_VERSION_TYPE`?

Reply:
There was `('55.0a1', '55.0b1'),` hidden above. I reordered the cases and added some more to be more explicit for future readers 😃